### PR TITLE
get pvc again to get the pv name that bound to the PVC

### LINKED ISSUE #116515: [Failing Job] pull-kubernetes-e2e-gce-cos-alpha-features
Seen in pull-kubernetes-e2e-gce-cos-alpha-features

https://testgrid.k8s.io/presubmits-kubernetes-nonblocking#pull-kubernetes-e2e-gce-cos-alpha-features

Specific job failure

https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/115075/pull-kubernetes-e2e-gce-cos-alpha-features/1635022577460056064/

```
I0312 21:34:53.652]   [FAILED] failed to get pv , resource name may not be empty
I0312 21:34:53.652]   In [It] at: test/e2e/storage/csi_mock/csi_volume_expansion.go:279 @ 03/12/23 21:33:48.709
I0312 21:34:53.652] ------------------------------
I0312 22:03:53.663] SSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSS•SSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSS•SSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSS•SSSSSSSSSSSSSSSSSSSSSSS•SSSSSSSSSSSSSSSS•SSSSSSSSSSSS•SSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSS•SSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSS•SSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSS•SSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSS•SSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSS•SSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSS•SSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSS•S•SSSSSSSSSSSSSSSSS•SSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSS•SSSSSSSSSSSSSSSSSSSS•SSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSS•SSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSS•SSSSSSSSSSS•S•SSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSS•SSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSS•SSSSSSSSSSSSSSSSSSSSSSSSSSSSSS•SSSSSSSSSSSSSSSSSS•SSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSS•SSSSSSSSSSSSSS•SSSSSSSSSSSSSSSSSSSSSSSS•SSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSS•SSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSS•SSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSS•SSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSS•SSSSSSS•SSSSS•SSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSS•SSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSS•SSSSSSSSSSSSSSSSSSSSSSSSSSSSS•SSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSS•SSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSS•SSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSS•SSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSGinkgo detected an issue with your spec structure
I0312 22:03:53.680] test/e2e/auth/selfsubjectreviews.go:50
I0312 22:03:53.680]   It looks like you are trying to add a [Container] node
I0312 22:03:53.680]   to the Ginkgo spec tree in a leaf node after the specs started running.
I0312 22:03:53.680] 
I0312 22:03:53.680]   To enable randomization and parallelization Ginkgo requires the spec tree
I0312 22:03:53.680]   to be fully constructed up front.  In practice, this means that you can
I0312 22:03:53.680]   only create nodes like [Container] at the top-level or within the
I0312 22:03:53.680]   body of a Describe, Context, or When.
I0312 22:03:53.680] 
I0312 22:03:53.680]   Learn more at:
I0312 22:03:53.680]   http://onsi.github.io/ginkgo/#mental-model-how-ginkgo-traverses-the-spec-hierarchy
I0312 22:03:53.680] 
I0312 22:03:53.681] 
I0312 22:03:53.690] Ginkgo ran 1 suite in 39m53.652188742s
I0312 22:03:53.691] 
I0312 22:03:53.691] Test Suite Failed
```

### PR DESCRIPTION
**Pull request by pacoxu**
#### What type of PR is this?

/kind failing-test
/kind flake

#### Which issue(s) this PR fixes:

Fixes #116515

#### Special notes for your reviewer:
Following https://github.com/kubernetes/kubernetes/issues/116515#issuecomment-1467412338.
#### Does this PR introduce a user-facing change?

```release-note
None
```

### CODE REVIEW

File: test/e2e/storage/csi_mock/csi_volume_expansion.go [hunk 1]
```diff
@@ -274,6 +274,10 @@ var _ = utils.SIGDescribe("CSI Mock volume expansion", func() {
 				err = e2epod.WaitForPodNameRunningInNamespace(ctx, m.cs, pod.Name, pod.Namespace)
 				framework.ExpectNoError(err, "Failed to start pod1: %v", err)
 
+				pvc, err = m.cs.CoreV1().PersistentVolumeClaims(pvc.Namespace).Get(ctx, pvc.Name, metav1.GetOptions{})
```
Comment (msau42) [inline test/e2e/storage/csi_mock/csi_volume_expansion.go:277]: Can you also remove the "Feature" tag from this test? The feature flag was promoted to beta so the test can run in the default jobs now: https://github.com/kubernetes/kubernetes/pull/115621

Comment (pacoxu) [reply test/e2e/storage/csi_mock/csi_volume_expansion.go:277]: Updated.

I also add `pvc.Spec.VolumeName` check before getting the PV.
```
 gomega.Expect(pvc.Spec.VolumeName).ShouldNot(gomega.BeEquivalentTo(""))
```

File: test/e2e/storage/csi_mock/csi_volume_expansion.go [hunk 2]
```diff
@@ -388,7 +393,7 @@ var _ = utils.SIGDescribe("CSI Mock volume expansion", func() {
 		}
 	})
 
-	ginkgo.Context("Expansion with recovery[Feature:RecoverVolumeExpansionFailure]", func() {
+	ginkgo.Context("Expansion with recovery", func() {
```
Comment (msau42) [inline test/e2e/storage/csi_mock/csi_volume_expansion.go:396]: Wrong feature tag. It should be the one on L204: "[Feature:CSINodeExpandSecret]"

File: test/e2e/storage/csi_mock/csi_volume_expansion.go [hunk 3]
```diff
@@ -274,6 +274,11 @@ var _ = utils.SIGDescribe("CSI Mock volume expansion", func() {
 				err = e2epod.WaitForPodNameRunningInNamespace(ctx, m.cs, pod.Name, pod.Namespace)
 				framework.ExpectNoError(err, "Failed to start pod1: %v", err)
 
+				pvc, err = m.cs.CoreV1().PersistentVolumeClaims(pvc.Namespace).Get(ctx, pvc.Name, metav1.GetOptions{})
+				if err != nil {
+					framework.Failf("failed to get pvc %s, %v", pvc.Name, err)
+				}
+				gomega.Expect(pvc.Spec.VolumeName).ShouldNot(gomega.BeEquivalentTo(""))
```
Comment (msau42) [inline test/e2e/storage/csi_mock/csi_volume_expansion.go:281]: Can you add some context at the end, for example, "while provisioning a volume for resizing"

Comment (pacoxu) [reply test/e2e/storage/csi_mock/csi_volume_expansion.go]: Done.

**Comment by pacoxu** [comment]
It failed below:
```

Kubernetes e2e suite: [It] [sig-storage] CSI Mock volume expansion CSI online volume expansion with secret should expand volume without restarting pod if attach=on, nodeExpansion=on, csiNodeExpandSecret=on expand_less	1m55s

{
 failed [FAILED] while waiting for all CSI calls: Unexpected secret: expected map[password:t0p-Secret username:admin], got map[]
In [It] at: test/e2e/storage/csi_mock/csi_volume_expansion.go:322 @ 03/14/23 06:57:33.822
}

```

**Comment by msau42** [comment]
I don't think this will fix the error, but since this feature is testing the NodeExpandVolume call, I think this check needs to move to after `WaitForFSResize`:
https://github.com/kubernetes/kubernetes/blob/d42885a2300706ba322a637a4e845f56b72e92d0/test/e2e/storage/csi_mock/csi_volume_expansion.go#L306-L322

File: test/e2e/storage/csi_mock/csi_volume_expansion.go [hunk 4]
```diff
@@ -274,6 +274,11 @@ var _ = utils.SIGDescribe("CSI Mock volume expansion", func() {
 				err = e2epod.WaitForPodNameRunningInNamespace(ctx, m.cs, pod.Name, pod.Namespace)
 				framework.ExpectNoError(err, "Failed to start pod1: %v", err)
 
+				pvc, err = m.cs.CoreV1().PersistentVolumeClaims(pvc.Namespace).Get(ctx, pvc.Name, metav1.GetOptions{})
+				if err != nil {
+					framework.Failf("failed to get pvc %s, %v", pvc.Name, err)
+				}
+				gomega.Expect(pvc.Spec.VolumeName).ShouldNot(gomega.BeEquivalentTo(""), "while provisioning a volume for resizing")
 				pv, err := m.cs.CoreV1().PersistentVolumes().Get(ctx, pvc.Spec.VolumeName, metav1.GetOptions{})
```
Comment (msau42) [inline test/e2e/storage/csi_mock/csi_volume_expansion.go:282]: To try to debug the test some more, we can also add validation that the `PV.spec.csi.nodeExpandSecretRef` is set correctly.

**Comment by zhucan** [comment]
> 

move to after WaitForFSResize no any effects.

File: test/e2e/storage/csi_mock/csi_volume_expansion.go (hunk 4 above)
Comment (pacoxu) [reply test/e2e/storage/csi_mock/csi_volume_expansion.go:282]: Added:
https://github.com/kubernetes/kubernetes/pull/116568/files#diff-6418fc90a4ed7e7c3cbfe9119ad04e252cf38d9ff1be39fcde2eb7f41178d641R289-R291

**Comment by pacoxu** [comment]
> I don't think this will fix the error, but since this feature is testing the NodeExpandVolume call, I think this check needs to move to after `WaitForFSResize`.

Updated.

**Comment by humblec** [comment]
> It failed below:
> 
> ```
> 
> Kubernetes e2e suite: [It] [sig-storage] CSI Mock volume expansion CSI online volume expansion with secret should expand volume without restarting pod if attach=on, nodeExpansion=on, csiNodeExpandSecret=on expand_less	1m55s
> 
> {
>  failed [FAILED] while waiting for all CSI calls: Unexpected secret: expected map[password:t0p-Secret username:admin], got map[]
> In [It] at: test/e2e/storage/csi_mock/csi_volume_expansion.go:322 @ 03/14/23 06:57:33.822
> }
> ```

Looks like this is what causing the latest failure.. that said, the test use external provisioner v3.3.0 which has `csiNodeExpandSecret` feature in alpha and default to OFF..  even the latest version of the provisioner 3.4.0 has the feature in the same status.. 
so we have to enable the provisioner flag explicitly to ON and check the result..

**Comment by zhucan** [comment]
@pacoxu  The csi provisioner which is in use is 3.3.0, the feature gate is alpha, We need to enable it .https://github.com/kubernetes/kubernetes/blob/d42885a2300706ba322a637a4e845f56b72e92d0/test/e2e/testing-manifests/storage-csi/mock/csi-mock-driver.yaml#L23

File: test/e2e/storage/csi_mock/csi_volume_expansion.go [hunk 5]
```diff
@@ -274,13 +274,21 @@ var _ = utils.SIGDescribe("CSI Mock volume expansion", func() {
...
 				if err != nil {
 					framework.Failf("failed to get pv %s, %v", pvc.Spec.VolumeName, err)
 				}
 				if pv.Spec.CSI == nil || pv.Spec.CSI.NodeExpandSecretRef == nil {
 					framework.Fail("creating pv without 'NodeExpandSecretRef'")
 				}
+				if pv.Spec.CSI.NodeExpandSecretRef.Namespace != csiNodeExpandSecretNamespaceKey || pv.Spec.CSI.NodeExpandSecretRef.Name != csiNodeExpandSecretKey {
```
Comment (zhucan) [inline test/e2e/storage/csi_mock/csi_volume_expansion.go:289]:
```suggestion
				if pv.Spec.CSI.NodeExpandSecretRef.Namespace != f.Namespace.Name || pv.Spec.CSI.NodeExpandSecretRef.Name != secretName {
```

Comment (zhucan) [reply test/e2e/storage/csi_mock/csi_volume_expansion.go:289]: image

Comment (pacoxu) [reply test/e2e/storage/csi_mock/csi_volume_expansion.go]: Update.

**Comment by pacoxu** [comment]
https://github.com/kubernetes/kubernetes/pull/116568/commits/98d5264245a007e1ab3ae83b4f7eb7e7ae286dad enabled the feature gate for csi provisioner.

**Comment by pacoxu** [comment]
```
W0314 08:11:15.223771       1 feature_gate.go:237] Setting GA feature gate Topology=true. It will be removed in a future release.
F0314 08:11:15.223859       1 csi-provisioner.go:138] unrecognized feature gate: CSINodeExpandSecret
```

I got this from the last run. CSINodeExpandSecret is not included in `registry.k8s.io/sig-storage/csi-provisioner:v3.3.0`?

**Comment by zhucan** [comment]
> 
https://github.com/kubernetes-csi/external-provisioner/commit/b416eb5a557c36fdb4390ead94708854a77f4ed9

**Comment by humblec** [comment]
> > 
> 
> [kubernetes-csi/external-provisioner@b416eb5](https://github.com/kubernetes-csi/external-provisioner/commit/b416eb5a557c36fdb4390ead94708854a77f4ed9)

@zhucan didnt there a feature flag in provisioner for this feature?

**Comment by humblec** [comment]
> > > 
> > 
> > 
> > [kubernetes-csi/external-provisioner@b416eb5](https://github.com/kubernetes-csi/external-provisioner/commit/b416eb5a557c36fdb4390ead94708854a77f4ed9)
> 
> @zhucan didnt there a feature flag in provisioner for this feature?

Looks like it is the case and causing this error https://github.com/kubernetes-csi/external-provisioner/blob/master/pkg/features/features.go .

**Comment by pacoxu** [comment]
According to https://github.com/kubernetes-csi/external-provisioner#feature-status, the `Provisioner Feature Gate Required` for CSINodeExpandSecret is No.

**Comment by humblec** [comment]
> https://storage.googleapis.com/kubernetes-jenkins/pr-logs/pull/116568/pull-kubernetes-e2e-kind/1635547325315158016/artifacts/csi-mock-volumes-expansion-2839-7205/csi-mockplugin-0/csi-provisioner/logs.txt
> 
> ```
> W0314 08:11:15.223771       1 feature_gate.go:237] Setting GA feature gate Topology=true. It will be removed in a future release.
> F0314 08:11:15.223859       1 csi-provisioner.go:138] unrecognized feature gate: CSINodeExpandSecret
> ```
> 
> I got this from the last run. CSINodeExpandSecret is not included in `registry.k8s.io/sig-storage/csi-provisioner:v3.3.0`?

@pacoxu looks like we have to remove the featuregate enablement commit from this PR and give a try again.

**Comment by humblec** [comment]
if the last run also fails on secret , for some reason the secret has not been refreshed on the csi mock method.. not sure just a GET on the created secret in the test case can help/validate.

**Comment by zhucan** [comment]
@pacoxu https://github.com/kubernetes/kubernetes/blob/921dfed168f1fbe7da697296604053c30a6000cd/test/e2e/storage/drivers/csi.go#L778 the secret comes from the mock csi logs, I'm not sure the "getCalls" whether it will refresh it.

**Comment by pacoxu** [comment]
> I0314 09:22:47.564460       1 server.go:101] GRPC call: /csi.v1.Node/NodeExpandVolume
> I0314 09:22:47.564490       1 server.go:105] GRPC request: {"capacity_range":{"required_bytes":6442450944},"secrets":"***stripped***","staging_target_path":"/var/lib/kubelet/plugins/kubernetes.io/csi/csi-mock-csi-mock-volumes-expansion-9751/8b55901b205257ba8f5953ac5bed8e6f70c4f182593b61eb3487f914dfd4f007/globalmount","volume_capability":{"AccessType":{"Mount":{"fs_type":"ext4"}},"access_mode":{"mode":7}},"volume_id":"8c4418e0-c249-11ed-b312-3e5752e18241","volume_path":"/var/lib/kubelet/pods/071edd65-ce1d-4258-96af-6f6d77d63dae/volumes/kubernetes.io~csi/pvc-0395fbb7-1166-46ef-b349-f921b8cdf4d7/mount"}
> I0314 09:22:47.564754       1 server.go:114] GRPC response: {}
> I0314 09:22:47.564825       1 server.go:159] gRPCCall: {"Method":"/csi.v1.Node/NodeExpandVolume","Request":{"volume_id":"8c4418e0-c249-11ed-b312-3e5752e18241","volume_path":"/var/lib/kubelet/pods/071edd65-ce1d-4258-96af-6f6d77d63dae/volumes/kubernetes.io~csi/pvc-0395fbb7-1166-46ef-b349-f921b8cdf4d7/mount","capacity_range":{"required_bytes":6442450944},"staging_target_path":"/var/lib/kubelet/plugins/kubernetes.io/csi/csi-mock-csi-mock-volumes-expansion-9751/8b55901b205257ba8f5953ac5bed8e6f70c4f182593b61eb3487f914dfd4f007/globalmount","volume_capability":{"AccessType":{"Mount":{"fs_type":"ext4"}},"access_mode":{"mode":7}},"secrets":{"password":"t0p-Secret","username":"admin"}},"Response":{},"Error":"","FullError":null}
> ==== end of pod log for container csi-mockplugin-0/mock ====

https://storage.googleapis.com/kubernetes-jenkins/pr-logs/pull/116568/pull-kubernetes-e2e-kind/1635564720662515712/artifacts/_sig-storage_CSI_Mock_volume_expansion/should_expand_volume_without_restarting_pod_if_attach_on_nodeExpansion_on_csiNodeExpandSecret_on/CSI_online_volume_expansion_with_secret/should_expand_volume_without_restarting_pod_if_attach_on_nodeExpansion_on_csiNodeExpandSecret_on/csi-mockplugin-0-mock.log

According to mock log, it seems the secrets is set correctly.

**Comment by pacoxu** [comment]
Let me add some extra logs about allCalls here.

**Comment by zhucan** [comment]
@pacoxu https://github.com/kubernetes/kubernetes/blob/c8f001d798eb81a92e61e467fca74ca180de6fc6/test/e2e/storage/drivers/csi.go#L382
`gRPCCall: {"Method":"/csi.v1.Node/NodeExpandVolume","Request":{"volume_id":"8c4418e0-c249-11ed-b312-3e5752e18241","volume_path":"/var/lib/kubelet/pods/071edd65-ce1d-4258-96af-6f6d77d63dae/volumes/kubernetes.io~csi/pvc-0395fbb7-1166-46ef-b349-f921b8cdf4d7/mount","capacity_range":{"required_bytes":6442450944},"staging_target_path":"/var/lib/kubelet/plugins/kubernetes.io/csi/csi-mock-csi-mock-volumes-expansion-9751/8b55901b205257ba8f5953ac5bed8e6f70c4f182593b61eb3487f914dfd4f007/globalmount","volume_capability":{"AccessType":{"Mount":{"fs_type":"ext4"}},"access_mode":{"mode":7}},"secrets":{"password":"t0p-Secret","username":"admin"}},"Response":{},"Error":"","FullError":null} `

please repplace the secret with secrets.

**Comment by zhucan** [comment]
> hooks is not nil; so the embedded is true. and the mock call filter logic will be skipped then.
> 
> ```
> 	if tp.hooks != nil {
> 		driverOpts.Embedded = true
> 		driverOpts.Hooks = *tp.hooks
> 	}
> 
> ...
> func (m *mockCSIDriver) GetCalls(ctx context.Context) ([]MockCSICall, error) {
> 	if m.embedded {
> 		return m.calls.Get(), nil
> 	}
> ...
> ```



> hooks is not nil; so the embedded is true. and the mock call filter logic will be skipped then.
> 
> ```
> 	if tp.hooks != nil {
> 		driverOpts.Embedded = true
> 		driverOpts.Hooks = *tp.hooks
> 	}
> 
> ...
> func (m *mockCSIDriver) GetCalls(ctx context.Context) ([]MockCSICall, error) {
> 	if m.embedded {
> 		return m.calls.Get(), nil
> 	}
> ...
> ```

The hooks is nil, so the  m.embedded  is false.

**Comment by zhucan** [comment]
> hooks is not nil; so the embedded is true. and the mock call filter logic will be skipped then.
> 
> ```
> 	if tp.hooks != nil {
> 		driverOpts.Embedded = true
> 		driverOpts.Hooks = *tp.hooks
> 	}
> 
> ...
> func (m *mockCSIDriver) GetCalls(ctx context.Context) ([]MockCSICall, error) {
> 	if m.embedded {
> 		return m.calls.Get(), nil
> 	}
> ...
> ```

The hooks is nil, so the m.embedded is false.

**Comment by humblec** [comment]
the csi json tag is `secrets` as @zhucan mentioned in the log csi call output where the unmarshalling happens to `secret` , I think changing that as suggested should fix the issue. @pacoxu one more attempt with that fix ,  if you can ? :)

**Comment by humblec** [comment]
@pacoxu if you are around , can we pick the changes from https://github.com/kubernetes/kubernetes/pull/116578 where tests are passing !!  ?

**Comment by pacoxu** [comment]
I cherry-picked the commit from @zhucan 's commit to this one.

File: test/e2e/storage/csi_mock/csi_volume_expansion.go [hunk 6]
```diff
@@ -201,7 +201,7 @@ var _ = utils.SIGDescribe("CSI Mock volume expansion", func() {
 			})
 		}
 	})
-	ginkgo.Context("CSI online volume expansion with secret[Feature:CSINodeExpandSecret]", func() {
+	ginkgo.Context("CSI online volume expansion with secret", func() {
 		var stringSecret = map[string]string{
```
Comment (gnufied) [inline test/e2e/storage/csi_mock/csi_volume_expansion.go:205]: nit: should say `node secret`

**Pull request merged**